pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -230,6 +236,20 @@ func (p *DefaultProvider) createLaunchTemplate(ctx context.Context, options *ami
 		LaunchTemplateName: aws.String(LaunchTemplateName(options)),
 		LaunchTemplateData: &ec2types.RequestLaunchTemplateData{
 			BlockDeviceMappings: p.blockDeviceMappings(options.BlockDeviceMappings),
+			CapacityReservationSpecification: &ec2types.LaunchTemplateCapacityReservationSpecificationRequest{
+				CapacityReservationPreference: lo.Ternary(
+					options.CapacityType == karpv1.CapacityTypeReserved,
+					ec2types.CapacityReservationPreferenceCapacityReservationsOnly,
+					ec2types.CapacityReservationPreferenceNone,
```

Comment:
If we are treating this as something that is under a feature flag -- this technically change behavior since the current default for this is `open` (though, I'm not sure how much this tangibly matters since I think it's fairly unlikely that you're getting lucky enough to match open reservations -- interested to hear your thoughts, though